File: src/main/resources/Authentication.py
from com.fererlab.service import Service


class Authentication(Service):
    def handle(self, event):
        print("Python object: {}".format(self))
        print("Python Event: {}".format(event))
        print("username: {}".format(event.body["username"]))
        return {"logged": True, "groups": ["admin", "user"]}

    def sayHi(self, request):
        print("Python object: {}".format(self))
        print("Python Request: {}".format(request))
        return {"say": "Hi {}".format(request.params["name"])}

File: src/main/resources/Authentication.rb
class Authentication
  include com.fererlab.service.Service

  def handle(event)
    puts "Ruby Object: #{self}"
    puts "Ruby Event: #{event}"
    puts "username: #{event.body["username"]}"
    Hash[:logged => true, :groups => %w(admin user)]
  end

  def sayHi(request)
    puts "Ruby Object: #{self}"
    puts "Ruby Request: #{request}"
    Hash[:say => "Hi #{request.params["name"]}"]
  end

end
AuthenticationG.new
